pyomo/contrib/alternative_solutions/aos_utils.py

```diff
@@ -0,0 +1,290 @@
+    else:
+        for comp in components:
+            if hasattr(comp, "ctype") and comp.ctype == pe.Block:
+                blocks = comp.values() if comp.is_indexed() else (comp,)
+                for item in blocks:
+                    variables = vfe.get_vars_from_components(
+                        item, component_list, include_fixed=include_fixed
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif (
+                isinstance(comp, tuple)
+                and hasattr(comp[0], "ctype")
+                and comp[0].ctype == pe.Block
+            ):
+                block = comp[0]
+                descend_into = pe.Block if comp[1] else False
+                blocks = block.values() if block.is_indexed() else (block,)
+                for item in blocks:
+                    variables = vfe.get_vars_from_components(
+                        item,
+                        component_list,
+                        include_fixed=include_fixed,
+                        descend_into=descend_into,
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif hasattr(comp, "ctype") and comp.ctype in component_list:
+                constraints = comp.values() if comp.is_indexed() else (comp,)
+                for item in constraints:
+                    variables = pe.expr.identify_variables(
+                        item.expr, include_fixed=include_fixed
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif hasattr(comp, "ctype") and comp.ctype == pe.Var:
+                variables = comp.values() if comp.is_indexed() else (comp,)
+                _filter_model_variables(
+                    variable_set,
+                    variables,
+                    include_continuous,
+                    include_binary,
+                    include_integer,
+                    include_fixed,
+                )
+            else:  # pragma: no cover
+                if not quiet:
+                    print(
+                        ("No variables added for unrecognized component {}.").format(
+                            comp
+                        )
+                    )
```

Comment:
In principle, I'm OK with this.  But I don't know how to use the Pyomo logger.  Is it as easy as passing-in a boolean flag?

I'm inclined to make this change later, since I wouldn't know how to implement or test it.

Reply:
It's pretty easy--mostly a find and replace of print statements, though you'll have to choose the logging level for them. We usually just do something along the lines of:
```
import logging

logger = logging.getLogger(__name__)
```
at the top of the file. 

Then, instead of print statements, you can just do: `logger.warn("No variables added...")` or `logger.debug("Starting No-Good cut analysis")`, etc. Then users can just set the logging level to what they want, direct it to a file instead of stdout, or whatever, and you don't have to handle any of that logic. The levels and things are in [these docs](https://docs.python.org/3/howto/logging.html)

And if you want to test it, you can use `pyomo.common.log.LoggingIntercept`, but I think that can definitely wait until after this PR for that.

Reply:
Done